### PR TITLE
Add stubs for rt_sigaction and rt_sigprocmask in SGX syscall handler

### LINKED ISSUE #403: Add rt_sig* stubs for SGX syscall handler
Since the enclave never receives these signals, these can just return 0.

### PR DESCRIPTION
**Pull request by lkatalin**
Resolves #403 

Related debt issue is #405

### CODE REVIEW

File: enarx-keep-sgx-shim/src/handler.rs [hunk 1]
```diff
@@ -191,4 +191,18 @@ impl<'a> Handler<'a> {
             ret
         }
     }
+
+    /// rt_sigaction() syscall stub
+    /// This type of signal is not delivered to the enclave, so a dummy value
```
Comment (haraldh) [inline enarx-keep-sgx-shim/src/handler.rs:196]: Hmm, the comment is kind of wrong.

Even though, the outside world does not send a signal to the enclave process, the enclave could use `kill()` to send itsself a signal.

We also have all other kind of signal, see e.g. the `si_code` field documentation in the `rt_sigaction` man page.

File: enarx-keep-sgx-shim/src/handler.rs [hunk 2]
```diff
@@ -191,4 +191,18 @@ impl<'a> Handler<'a> {
...
+    /// is returned.
+    pub fn rt_sigaction(&mut self) -> u64 {
+        0
+    }
+
+    /// rt_sigprocmask() syscall stub
+    /// This type of signal is not delivered to the enclave, so a dummy value
+    /// is returned.
```
Comment (haraldh) [inline enarx-keep-sgx-shim/src/handler.rs:204]: same as for rt_sigaction.

**Comment by npmccallum** [comment]
I agree with @haraldh's comments. Address those and I'll approve.

**Comment by lkatalin** [comment]
@npmccallum @haraldh I updated the comments and filed a debt issue: #405 .

**Comment by npmccallum** [comment]
Superseded by #408.

**Pull request closed without merging**